### PR TITLE
feat: add decay_params method

### PR DESCRIPTION
**Pull request by CjS77**
Wondering how the emission schedule parameters are derived?

This provides a `decay_params` method that calculates the decay array parameters
used in EmissionSchedule.

The method serves as reference and means of determining the array
parameters.

### CODE REVIEW

File: base_layer/core/src/consensus/emission.rs [hunk 1]
```diff
@@ -73,6 +73,67 @@ impl EmissionSchedule {
...
+    /// (1-k)*n (to 1/2^64 precision).
+    ///
+    /// None - If k is not a valid floating point number less than one.
+    pub fn decay_params(k: &str) -> Option<Vec<u8>> {
+        // Convert string into a vector of digits. e.g. 0.9635 -> [9,6,3,5]
+        fn frac_vec(n: &str) -> Option<Vec<u8>> {
+            if !n.starts_with("0.") {
+                return None;
+            }
+            if n.chars().skip(2).any(|i| i < '0' || i > '9') {
```
Comment (sdbondi) [inline base_layer/core/src/consensus/emission.rs]: 📎 
```suggestion
            if n.chars().skip(2).any(|i| ('0'..='9').contains(&i)) {
```

**Review by sdbondi — DISMISSED** [review]
Nice, should help demystify the "magic" decay values. Just one 📎

**Review by sdbondi — DISMISSED** [review]
LGTM

**Comment by delta1** [comment]
It seems there might be a test that needs updating for this @CjS77

**Comment by CjS77** [comment]
Tests are fine. One of the suggestions actually broke the code. Will push up fix now.

**Review by delta1 — APPROVED** [review]

**Review by stringhandler — APPROVED** [review]

**Pull request merged**